baybe/campaign.py

```diff
@@ -564,30 +570,133 @@ def get_surrogate(
                 f"No surrogate is available since no '{Objective.__name__}' is defined."
             )
 
-        recommender: RecommenderProtocol
-        if isinstance(self.recommender, MetaRecommender):
-            recommender = self.recommender.get_non_meta_recommender(
-                batch_size,
-                self.searchspace,
-                self.objective,
-                self.measurements,
-                pending_experiments,
-            )
-        else:
-            recommender = self.recommender
-
+        recommender = self._get_non_meta_recommender(batch_size, pending_experiments)
         if isinstance(recommender, BayesianRecommender):
             return recommender.get_surrogate(
                 self.searchspace, self.objective, self.measurements
             )
         else:
-            raise RuntimeError(
+            raise IncompatibilityError(
                 f"The current recommender is of type "
                 f"'{recommender.__class__.__name__}', which does not provide "
                 f"a surrogate model. Surrogate models are only available for "
                 f"recommender subclasses of '{BayesianRecommender.__name__}'."
             )
 
+    def _get_non_meta_recommender(
+        self,
+        batch_size: int | None = None,
+        pending_experiments: pd.DataFrame | None = None,
+    ) -> RecommenderProtocol:
+        """Get the current recommender.
+
+        Args:
+            batch_size: See :meth:`recommend`.
+                Only required when using meta recommenders that demand it.
+            pending_experiments: See :meth:`recommend`.
+                Only required when using meta recommenders that demand it.
+
+        Returns:
+            The recommender for the current recommendation context.
+        """
+        if not isinstance(self.recommender, MetaRecommender):
+            return self.recommender
+        return self.recommender.get_non_meta_recommender(
+            batch_size,
+            self.searchspace,
+            self.objective,
+            self.measurements,
+            pending_experiments,
+        )
+
+    @overload
+    def acquisition_values(
+        self,
+        candidates: pd.DataFrame | None = None,
+        *,
+        joint: Literal[False],
+        batch_size: int | None = None,
+        pending_experiments: pd.DataFrame | None = None,
+    ) -> pd.Series: ...
+
+    @overload
+    def acquisition_values(
+        self,
+        candidates: pd.DataFrame | None = None,
+        *,
+        joint: Literal[True],
+        batch_size: int | None = None,
+        pending_experiments: pd.DataFrame | None = None,
+    ) -> float: ...
+
+    def acquisition_values(
```

Comment:
@Scienfitz: The user guide section is still missing. The reason why I haven't added it yet is because I first wanted to hear your opinion about how we want to design things here. Right now, this is implemented as a `dataframe-based` method. However, with #504, we now basically have both the complete torch and dataframe versions for the posterior. What do you think, how should we handle it here?

Reply:
I think the torch part became sort of obsolet now because I've further added an explicit `get_acquisition_function` convenience method, which gives direct access to the botorch object. But the user guide is still missing. Will wait for a decision from [here](https://github.com/emdgroup/baybe/pull/509#discussion_r1998274529)

Reply:
Since there is now [this](https://github.com/emdgroup/baybe/pull/509#discussion_r2041988076) other thread about docs, I'll close here